File: typescript/src/cli/utils/file-ops.ts
/**
 * File operations for the Mirascope CLI.
 */

import { mkdir } from "fs/promises";
import { dirname } from "path";

export async function writeFile(path: string, content: string): Promise<void> {
  // Create parent directories if they don't exist
  const dir = dirname(path);
  await mkdir(dir, { recursive: true });

  // Write the file
  // eslint-disable-next-line no-undef
  await Bun.write(path, content);
}

export async function ensureDirectory(path: string): Promise<void> {
  await mkdir(path, { recursive: true });
}

File: python/mirascope/cli/utils/file_ops.py
"""File operations for the Mirascope CLI."""

from __future__ import annotations

from pathlib import Path


def write_file(path: Path, content: str) -> None:
    """Write content to a file, creating directories if needed.

    Args:
        path: Path to write to.
        content: Content to write.
    """
    # Create parent directories if they don't exist
    path.parent.mkdir(parents=True, exist_ok=True)

    # Write the file
    path.write_text(content)


def ensure_directory(path: Path) -> None:
    """Ensure a directory exists.

    Args:
        path: Directory path to ensure exists.
    """
    path.mkdir(parents=True, exist_ok=True)
